Runtime/AvatarCreator/UI/PhotoCaptureElement.cs

```diff
@@ -0,0 +1,79 @@
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+    public class PhotoCaptureElement : MonoBehaviour
+    {
+        [Header("Settings")]
+        [SerializeField] private RawImage cameraTextureTarget;
+
+        [Space(5)]
+        [Header("Events")]
+        public UnityEvent<Texture2D> onPhotoCaptured;
+
+        private WebCamTexture cameraTexture;
+
+        private void Awake()
+        {
+            InitializeCamera();
+        }
+
+        public void StartCamera()
+        {
+            if (cameraTexture != null && !cameraTexture.isPlaying)
+            {
+                cameraTexture.Play();
+            }
+        }
+
+        public void StopCamera()
+        {
+            if (cameraTexture != null && cameraTexture.isPlaying)
+            {
+                cameraTexture.Stop();
+            }
+        }
+
+        public void TakePhoto()
+        {
+            if (cameraTexture == null || !cameraTexture.isPlaying)
+                return;
+
+            var texture = new Texture2D(cameraTextureTarget.texture.width, cameraTextureTarget.texture.height, TextureFormat.ARGB32, false);
+            texture.SetPixels(cameraTexture.GetPixels());
+            texture.Apply();
+
+            onPhotoCaptured?.Invoke(texture);
+        }
+
+        private void InitializeCamera()
+        {
+            var webCamDevice = GetWebCamDevice();
+
+            SetupPhotoBoothTexture(webCamDevice?.name);
+            StartCamera();
```

Comment:
Actually not relevant to this PR, something I kinda missed in Max's PR. But perhaps we shouldn't force them to start the camera on awake (this InitializeCamera function is called on awake, it is also kinda a hidden "side effect" of initializing. I also wonder if it the automatic GetCamDeveice() on start might be annoying for some as it basically requests access regardless of whether the element is actually used? 🤔  Maybe I'm just overthinking this part though.

Reply:
I could see the positives for moving out the StartCamera() method to be invoked outside of awake (maybe in a from the editor itself in the element prefab), but then you'd maybe want the whole initialize method to not be on awake right?

Although since this is so easy to change between, maybe it's one of those ones where it's a case of us getting it out there and seeing what partners say. If we get reports that it is annoying then it's easy for us to change it and to respond to feedback.